core/capabilities/ccip/ccipevm/helpers.go

```diff
@@ -43,6 +49,42 @@ func decodeExtraArgsV1V2(extraArgs []byte) (gasLimit *big.Int, err error) {
 	return ifaces[0].(*big.Int), nil
 }
 
+func DecodeExtraArgsToMap(extraArgs []byte) (map[string]any, error) {
+	if len(extraArgs) < 4 {
+		return nil, fmt.Errorf("extra args too short: %d, should be at least 4 (i.e the extraArgs tag)", len(extraArgs))
+	}
+
+	var method string
+	var extraByteOffset int
+	switch string(extraArgs[:4]) {
+	case string(evmExtraArgsV1Tag):
+		// for EVMExtraArgs, the first four bytes is the method name
+		method = evmV1DecodeName
+		extraByteOffset = 4
+	case string(evmExtraArgsV2Tag):
+		method = evmV2DecodeName
+		extraByteOffset = 4
+	case string(svmExtraArgsV1Tag):
+		// for SVMExtraArgs there's the four bytes plus another 32 bytes padding for the dynamic array
+		method = svmV1DecodeName
+		extraByteOffset = 36
```

Comment:
I believe this is no longer needed because common would call the Solana version of this function directly.

Reply:
This is for the case when source chain is EVM, and we need ABI decoding instead of Borsh

Reply:
Could you add some comments with pointers to the onchain code?[ i.e. something like we've done here.](https://github.com/smartcontractkit/chainlink-ccip/blob/837b0a4b259955f3bc0c122d636221b806e9ffed/pkg/reader/ccip.go#L631)